superset-frontend/src/SqlLab/components/ExploreCtasResultsButton/index.jsx

```diff
@@ -35,29 +35,24 @@ const propTypes = {
   templateParams: PropTypes.string,
 };
 
-class ExploreCtasResultsButton extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.visualize = this.visualize.bind(this);
-    this.onClick = this.onClick.bind(this);
-  }
-
-  onClick() {
-    this.visualize();
-  }
-
-  buildVizOptions() {
-    return {
-      datasourceName: this.props.table,
-      schema: this.props.schema,
-      dbId: this.props.dbId,
-      templateParams: this.props.templateParams,
-    };
-  }
+function ExploreCtasResultsButton({
+  table,
+  schema,
+  dbId,
+  templateParams,
+  errorMessage,
+  actions,
+}) {
+  const { createCtasDatasource, addInfoToast, addDangerToast } = actions;
+  const buildVizOptions = {
+    datasourceName: table,
+    schema,
+    dbId,
+    templateParams,
+  };
 
-  visualize() {
-    this.props.actions
-      .createCtasDatasource(this.buildVizOptions())
+  const visualize = () => {
+    createCtasDatasource(buildVizOptions())
```

Comment:
looks like buildVizOptions is no longer a function?

Reply:
Ah! Yes. We were thinking that since the original buildVizOptions just returns an object, why not just create that object, save it as a variable, then pass it as needed. Maybe I should have left it as a function?

Reply:
AH! I get it. I should remove the "()" after buildVizOptions on line 55 then if we want it to just be an object. Or make buildVizOptions back into a function. Which would you prefer?